crates/re_renderer/src/wgpu_resources/mod.rs

```diff
@@ -163,4 +186,40 @@ impl TextureRowDataInfo {
 
         unpadded_buffer.into()
     }
+
+    /// Removes the padding from a buffer containing gpu texture data and remove convert to a given type.
+    ///
+    /// The buffer have the expected size for a padded buffer to hold the texture data.
+    ///
+    /// The unpadded row size is expected to be a multiple of the size of the target type.
+    /// (Which means that, while uncommon, it technically doesn't need to be as big as a block in the pixel - this can be useful for e.g. packing wide bitfields)
+    pub fn remove_padding_and_convert<T: bytemuck::Pod>(&self, buffer: &[u8]) -> Vec<T> {
+        crate::profile_function!();
+
+        assert!(buffer.len() as wgpu::BufferAddress == self.buffer_size_padded);
+        assert!(self.bytes_per_row_unpadded % std::mem::size_of::<T>() as u32 == 0);
+
+        // Due to https://github.com/gfx-rs/wgpu/issues/3508 the data might be completely unaligned,
+        // so much, that we can't even interpret it as e.g. a u32 slice.
+        // Therefore, we have to do a copy of the data regardless of whether it's padded or not.
+
+        let mut unpadded_buffer: Vec<T> = vec![
+            T::zeroed();
+            (self.num_rows() * self.bytes_per_row_unpadded / std::mem::size_of::<T>() as u32)
+                as usize
+        ]; // Consider using unsafe set_len() instead of vec![] to avoid zeroing the memory.
+        let unpaadded_buffer_u8 = bytemuck::cast_slice_mut(&mut unpadded_buffer);
```

Comment:
Typo + letting type inference decides the output of a transmutation makes me feel very uneasy :grimacing: 
```suggestion
        let unpadded_buffer_u8: &[u8] = bytemuck::cast_slice_mut(&mut unpadded_buffer);